### PR TITLE
I made my homework

### PR DESCRIPTION
**Pull request by Kani103**

### CODE REVIEW

File: week2/homework/src/doList.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+  "1": "Do your homework buddy!",
+  "2": "Practise to master it.",
+  "3": "coding with love my friend"
+}
```
Comment (eidosam) [inline week2/homework/src/doList.json]: This file is not used, so you need to replace it with the file you are using ( `todoList.txt`)

File: week2/homework/src/index.js [hunk 2]
```diff
@@ -1,3 +1,78 @@
 'use strict';
 
 // TODO: Write the homework code in this file
+let fs = require ('fs');
```
Comment (eidosam) [inline week2/homework/src/index.js]: Variable `fs` is not re-assigned, so it has to be declared with `const`

File: week2/homework/src/index.js [hunk 3]
```diff
@@ -1,3 +1,78 @@
...
+console.log(process.argv);
```
Comment (eidosam) [inline week2/homework/src/index.js]: No need to print the process arguments, it will be confusing to user.

File: week2/homework/src/index.js [hunk 4]
```diff
@@ -1,3 +1,78 @@
...
+
+let args = process.argv.slice(2);
+//console.log(args);
+
```
Comment (eidosam) [inline week2/homework/src/index.js]: Unused code better to be removed.

File: week2/homework/src/index.js [hunk 5]
```diff
@@ -1,3 +1,78 @@
...
+   help(todoItem)
+} else if (command === 'add') {
+    add(todoItem)
+} else if (command === 'list') {
+    list(todoItem)
+} else if (command === 'remove') {
+    remove(todoItem)
+} else if (command === 'reset') {
+    reset(todoItem)
+}
```
Comment (eidosam) [inline week2/homework/src/index.js]: The case where user provides invalid command or no command should be considered.

File: week2/homework/src/index.js [hunk 6]
```diff
@@ -1,3 +1,78 @@
...
+    1_ help: to go to the instructions.
+    2_ add: to add items.
+    3_ list: to show the list of the items
+    4_ remove: to remove an item
+    5_ reset: to delete all the item(the list)`)
+}
+
+function add (todoItem) {
+    fs.appendFile('./todoList.txt', todoItem + '\n', (error) => {
+        console.error(error)
```
Comment (eidosam) [inline week2/homework/src/index.js]: Better to check the error and print it only if it exists.

File: week2/homework/src/index.js [hunk 7]
```diff
@@ -1,3 +1,78 @@
...
+
+function remove (todoItem) {
+    fs.readFile('./todoList.txt', 'utf8', (error, todoList) => {
+        if (error) {
+            console.log(error)
+        } else {
+            let newText = todoList.split('\n');
+            newText.splice(todoItem -1, 1)
+            fs.writeFile('./todoList.txt', newText.join('\n'), (error) => {
+                console.log(error)
```
Comment (eidosam) [inline week2/homework/src/index.js]: Same note about checking the error.

File: week2/homework/src/index.js [hunk 8]
```diff
@@ -1,3 +1,78 @@
 'use strict';
 
 // TODO: Write the homework code in this file
+let fs = require ('fs');
+console.log(process.argv);
+
+let args = process.argv.slice(2);
```
Comment (eidosam) [inline week2/homework/src/index.js]: Declare with `const`, because it is not re-assigned.

File: week2/homework/src/index.js [hunk 9]
```diff
@@ -1,3 +1,78 @@
...
+//console.log(args);
+
+let command = args[0];
+let todoItem = args[1]
+
```
Comment (eidosam) [inline week2/homework/src/index.js]: Same note about using `const`

File: week2/homework/src/index.js [hunk 10]
```diff
@@ -1,3 +1,78 @@
...
+    })
+}
+
+
+function remove (todoItem) {
+    fs.readFile('./todoList.txt', 'utf8', (error, todoList) => {
+        if (error) {
+            console.log(error)
+        } else {
+            let newText = todoList.split('\n');
```
Comment (eidosam) [inline week2/homework/src/index.js]: `const`

File: week2/homework/src/index.js [hunk 11]
```diff
@@ -1,3 +1,78 @@
...
+        }
+    })
+}
+
+function reset () {
+    fs.writeFile('./todoList.txt', '', function(error) {
+        if(error) {
+            console.log(error)
+        } else {
+            console.log('')
```
Comment (eidosam) [inline week2/homework/src/index.js]: Would be nice to show user the status or result of the command, lie _Your todo-list has been cleared/reset_

**Comment by wouterkleijn** [comment]
Thanks @eidosam for the comments. Please @Kani103 can you make sure to implement the feedback?

File: week2/homework/src/doList.json (hunk 1 above)
Comment (Kani103) [reply week2/homework/src/doList.json]: Hey, I didn't notice that when i pushed my homework. This file is of my old homework so i think they mixed up with the same old branch..

File: week2/homework/src/index.js (hunk 2 above)
Comment (Kani103) [reply week2/homework/src/index.js]: thanks for the advice..

File: week2/homework/src/index.js (hunk 4 above)
Comment (Kani103) [reply week2/homework/src/index.js]: i always forget that, i will keep it in mind.

File: week2/homework/src/index.js (hunk 5 above)
Comment (Kani103) [reply week2/homework/src/index.js]: i will correct it, thanks.

**Comment by Kani103** [comment]
Thanks eidosam and Wouter..

**Comment by wouterkleijn** [comment]
@Kani103 No problem. Did you push the necessary changes?

**Comment by Kani103** [comment]
Hey, i made the changes you can take a look.
thanks for the help..

**Pull request closed without merging**